### ISSUE TITLE
SyncClient hangs

### ISSUE THREAD

**Issue by hangj** [libsql/libsql-client-rs#48]
The async Client works just fine, but the following SyncClient hangs:

```rust
#[tokio::main(flavor = "current_thread")]
async fn main() -> anyhow::Result<()> {
    let db = libsql_client::SyncClient::from_env()?;
    let response = db.execute("SELECT * FROM users")?;

    println!("{:?}", response);

    Ok(())
}
```

**Comment by psarna** [comment]
Thanks for reporting @hangj! I see that SyncClient gets stuck with your example, and gets unblocked once I change it from `#[tokio::main(flavor = "current_thread")]` to just `#[tokio::main]`, so we probably block the current thread for something... I'll dig further

**Comment by psarna** [comment]
yeah, it's due to the fact we call `futures::executor::block_on` underneath, which clashes with single-threaded tokio runtime

**Comment by hangj** [comment]
Yeah, that's the reason.

But the real problem I think is that the `libsql_client::SyncClient` is not a real blocking client, I have to choose a runtime for it.
